### PR TITLE
Upgrade cms for new i18n backend

### PR DESCRIPTION
**Pull request by tractorcow**
Required for https://github.com/silverstripe/silverstripe-framework/pull/6558

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -6,7 +6,7 @@
 charset = utf-8
 end_of_line = lf
 indent_size = 4
-indent_style = tab
+indent_style = space
```
Comment (mikenz) [inline .editorconfig]: Yay! But was this really supposed to be in this PR?

Comment (tractorcow) [inline .editorconfig]: We should be using psr2 now; I think some upgrades were missing from cms module though. I should run a formatter over it.

Comment (tractorcow) [inline .editorconfig]: Oh look, we never merged it. =( https://github.com/silverstripe/silverstripe-cms/pull/1688

Comment (tractorcow) [inline .editorconfig]: Rebuilt at https://github.com/silverstripe/silverstripe-cms/pull/1727. I'll revert from this PR.

**Comment by tractorcow** [comment]
I've reverted any presumptuous psr2 reformatting on SiteTree.php; I'll wait until we merge the main psr2 PR instead :D

**Comment by tractorcow** [comment]
Ok, I've updated the tests and fixed some german localisations for plurals. Should be passing now. :D

note: CMS pr can be merged ahead of the framework, since it only fixes problems, it doesn't RELY on the new api for it to work.

**Comment by tractorcow** [comment]
This is ready to merge.

**Pull request merged**